cmd/traefik/traefik.go

```diff
@@ -248,10 +251,42 @@ func checkNewVersion() {
 	safe.Go(func() {
 		time.Sleep(10 * time.Minute)
 		version.CheckNewVersion()
-		for {
-			select {
-			case <-ticker.C:
-				version.CheckNewVersion()
+		for range ticker.C {
+			version.CheckNewVersion()
+		}
+	})
+}
+
+func stats(globalConfiguration *configuration.GlobalConfiguration) {
+	if globalConfiguration.SendAnonymousUsage {
+		log.Info(`
+Stats collection is enabled.
+Many thanks to contribute to Traefik improvement by allowing us to receive anonymous information from your configuration.
+Help us improve Traefik by leaving this feature on :)
+More details on: https://docs.traefik.io/basic/#collected-data
+`)
+		collect(globalConfiguration)
+	} else {
+		log.Info(`
+Stats collection is disabled.
+Help us improve Traefik by turning this feature on :)
+More details on: https://docs.traefik.io/basic/#collected-data
+`)
+	}
+}
+
+func collect(globalConfiguration *configuration.GlobalConfiguration) {
+	ticker := time.NewTicker(24 * time.Hour)
+	safe.Go(func() {
+		time.Sleep(10 * time.Minute)
+		err := collector.Collect(globalConfiguration)
+		if err != nil {
+			log.Debug(err)
+		}
+		for range ticker.C {
+			err := collector.Collect(globalConfiguration)
+			if err != nil {
+				log.Debug(err)
```

Comment:
It is possible to have just one call to `Collect` (slightly adopted from [here](https://www.reddit.com/r/golang/comments/5g0hor/ticker_with_instant_start/daov1qr/)):

```go
for time.Sleep(10 * time.Minute); ; <-ticker.C {
  if err := collector.Collect(globalConfiguration); err != nil {
    log.Debug(err)
  }
}
```

([Simplified Go Playground example](https://play.golang.org/p/xIhkaHd-Is).)